includes/admin/class-wp-job-manager-promoted-jobs.php

```diff
@@ -90,135 +93,34 @@ public function promoted_jobs_custom_columns( $column ) {
-			}
-			.promote-job-modal li.promote-list-item {
-				background: url("data:image/svg+xml,<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\"><mask id=\"mask0_20018663_2259\" style=\"mask-type:luminance\" maskUnits=\"userSpaceOnUse\" x=\"3\" y=\"5\" width=\"18\" height=\"14\"><path d=\"M8.75685 15.9L4.57746 11.7L3.18433 13.1L8.75685 18.7L20.698 6.69999L19.3049 5.29999L8.75685 15.9Z\" fill=\"white\"/></mask><g mask=\"url%28%23mask0_20018663_2259%29\"><rect width=\"23.8823\" height=\"24\" fill=\"%232270B1\"/></g></svg>") no-repeat 0 -3px;
-				font-size: 14px;
-				list-style: none;
-				padding-left: 32px;
-				margin: 12px 0;
-			}
-			.promote-job-modal .promote-job-modal-price {
-				display: flex;
-				flex-direction: column;
-			}
-			.promote-job-modal .promote-job-modal-price .price-text {
-				font-size: 12px;
-				text-transform: uppercase;
-				color: #787c82;
-			}
-			.promote-job-modal .promote-job-modal-price span {
-				margin-top: 10px;
-				font-size: 36px;
-				font-weight: 700;
-			}
-			.promote-job-modal .promote-buttons-group .button {
-				padding: 10px 16px;
-				border-radius: 2px;
-				margin-right: 18px;
-				border: 0;
-			}
-			.promote-job-modal .promote-buttons-group .button-primary {
-				background: #2270b1;
-				color: #fff;
-			}
-			.promote-job-modal .promote-buttons-group .button-secondary {
-				background: #fff;
-				color: #2270B1;
-				border: 1px solid #2270B1;
-			}
-			@media screen and (max-width: 1000px) {
-				.promote-job-modal-column-right {
-					display: none;
-				}
-			}
-			@media screen and (max-width: 782px) {
-				.promote-job-modal {
-					padding: 0px 25px 25px;
-				}
-			}
-			.wpjm-dialog {
-				border: 0;
-				border-radius: 8px;
-			}
-			form.dialog {
-				display: flex;
-			}
-			form.dialog button.dialog-close {
-				margin: 10px 15px auto auto;
-				content: "";
-				background: none;
-				border: 0;
-				font-size: 0;
-			}
-			form.dialog button.dialog-close:after {
-				content: "\2715";
-				font-size: 20px;
-			}
-		</style>
-		<div class="promote-job-modal">
-			<div class="promote-job-modal-column-left">
-				<h2 class="promote-jobs-heading">
-					Promote your job on our partner network.
-				</h2>
 
-				<div class="promote-job-modal-price">
-					<div class="price-text">Starting From</div>
-					<span>$83.00</span>
-				</div>
+		$promote_template = wp_cache_get( 'promote-jobs-template', 'promote-jobs', false, $found );
 
-				<ul class="promote-list">
-					<li class="promote-list-item">Your ad will get shared on our Partner Network</li>
-					<li class="promote-list-item">Featured on jobs.blog for 7 days</li>
-					<li class="promote-list-item">Featured on our weekly email blast</li>
-				</ul>
+		if ( ! $found ) {
+			$response         = wp_remote_get( 'http://wpjobmanager.com/wp-json/promoted-jobs/v1/assets/promote-dialog/?lang=' . get_locale() );
+			$promote_template = json_decode( $response['body'], true );
```

Comment:
In case `wp_remote_get`returns a WP_Error, could this line fail already? The `is_wp_error` check probably needs to be before an attempt to parse the response. 
Or could be safest to wrap this all in a try-catch :)

Reply:
I think we only need to update the cache if everything was successful, putting `wp_cache_set` after the 
```
$promote_template = json_decode( wp_remote_retrieve_body( $response ), true );
``` 
line. (And also caching the already parsed template instead of the response). 

That'd take care of this, falling back to the version we already cached. But I realized we'd need to change a bit more, de-coupling how often we check for new content from the cached content itself. We could have an option, something like `_promote-jobs-template_next_check`, that updates whenever we make a request, and the `promote-jobs-template` cache that never gets cleared unless we have a new version. (So maybe that too can be an option, not a cache.)

We'd make a new request if either time is up for the next check or the local template cache is empty.
For failed requests, we could make `_promote-jobs-template_next_check` to be a few minutes or hour from now, and for success it can be set longer.

Reply:
Great idea @yscik !

I think I covered most of this in f8e9194868bdfd1f9d1dbad22232c11f2a929873 but let me know if I missed something.

Reply:
Looks pretty good! But in case the update fails, I'd still fall back to the version saved in the `promote-jobs-template` option and return it in the `get_promote_jobs_template()` function — a slightly outdated version would still be good enough to show for this.

Reply:
Done in https://github.com/Automattic/WP-Job-Manager/pull/2478/commits/88e5c77b88005873e24d3e515a4ded2c2dbd0cec